src/lib/tdp-finder/index.ts

```diff
@@ -1,113 +1,124 @@
   data: any;
-  errorBuilder = buildErrorMessage(TdpFinderModel);
+  errorBuilder = buildErrorMessage(this.constructor);
+
+  /**
+   * Configures the TDP Finder Plugin.
+   */
+  public async configure(): Promise<ModelPluginInterface> {
+    this.data = await this.loadData();
+
+    return this;
+  }
 
   /**
    * Calculate the total emissions for a list of inputs.
-   *
-   * Each Input require:
-   * @param {Object[]} inputs
-   * @param {string} inputs[].timestamp RFC3339 timestamp string
    */
-  async execute(inputs: ModelParams[]): Promise<any[]> {
-    return inputs.map((input: KeyValuePair, index: number) => {
-      input['thermal-design-power'] = 0;
-      if ('physical-processor' in input) {
+  public async execute(inputs: ModelParams[]): Promise<ModelParams[]> {
+    return Promise.all(
+      inputs.map((input, index: number) => {
+        Object.assign(input, this.validateInput(input));
+
+        input['thermal-design-power'] = 0;
+
         const physicalProcessors = input['physical-processor'] as string;
-        physicalProcessors.split(',').forEach(physicalProcessor => {
-          physicalProcessor = physicalProcessor.trim();
-          if (
-            physicalProcessor in this.data &&
-            input['thermal-design-power'] < this.data[physicalProcessor]
-          ) {
-            input['thermal-design-power'] = this.data[physicalProcessor];
-          } else if (!(physicalProcessor in this.data)) {
+        const processors = physicalProcessors
+          .split(',')
+          .map(processor => processor.trim());
+
+        for (const processor of processors) {
+          if (processor in this.data) {
+            const currentTDP = input['thermal-design-power'];
+            input['thermal-design-power'] = Math.max(
+              currentTDP,
+              this.data[processor]
+            );
+          } else {
             throw new UnsupportedValueError(
               this.errorBuilder({
-                message: `'physical-processor': ${physicalProcessor} from input[${index}] is not found in database. Please check spelling / contribute to 'if-models' with the data.`,
+                message: `'physical-processor': ${processor} from input[${index}] is not found in the database`,
               })
             );
           }
-        });
-      } else {
-        throw new InputValidationError(
-          this.errorBuilder({
-            message: `'physical-processor' not provided in input[${index}].`,
-          })
-        );
-      }
-      return input;
-    });
-  }
-
-  async configure(
-    staticParams: object | undefined
-  ): Promise<ModelPluginInterface> {
-    this.staticParams = staticParams;
-    this.data = await this.loadData();
+        }
 
-    return this;
+        return input;
+      })
+    );
   }
 
-  async loadData(): Promise<any> {
+  /**
+   * Load data from csv files.
+   */
+  private async loadData(): Promise<KeyValuePair> {
     const data: KeyValuePair = {};
```

Comment:
if we have way to escape using key value pair

Reply:
done